src/main/java/teammates/sqllogic/core/UsageStatisticsLogic.java

```diff
@@ -0,0 +1,64 @@
+package teammates.sqllogic.core;
+
+import java.time.Instant;
+import java.util.List;
+
+import teammates.storage.sqlapi.UsageStatisticsDb;
+import teammates.storage.sqlentity.UsageStatistics;
+
+/**
+ * Handles operations related to system usage statistics objects.
+ *
+ * @see UsageStatistics
+ * @see teammates.storage.api.UsageStatisticsDb
+ */
+public final class UsageStatisticsLogic {
+
+    private static final UsageStatisticsLogic instance = new UsageStatisticsLogic();
+
+    private final UsageStatisticsDb usageStatisticsDb = UsageStatisticsDb.inst();
```

Comment:
Can pass in `usageStatisticsDb` as a dependency via `initLogicDependencies` method. This way, it's easier to mock this out if we need to do unit testing (for the more complex logic methods, once we move the logic out of the db layer).